tests/generation/test_utils.py

```diff
@@ -2346,21 +2427,22 @@ def _check_logits(self, batch_size, scores, config):
         self.assertTrue(vocab_diff in [0, 1])
         self.assertListEqual([vocab_size - score.shape[-1] for score in scores], [vocab_diff] * len(scores))
 
-    def _check_attentions_for_generate(
-        self, batch_size, attentions, min_length, max_length, config, use_cache=False, num_beam_groups=1
-    ):
+    def _check_attentions_for_generate(self, batch_size, attentions, min_length, max_length, config, past_key_values):
         self.assertIsInstance(attentions, tuple)
         self.assertListEqual(
             [isinstance(iter_attentions, tuple) for iter_attentions in attentions], [True] * len(attentions)
         )
-        self.assertEqual(len(attentions), (max_length - min_length) * num_beam_groups)
+        self.assertEqual(len(attentions), (max_length - min_length))
+
+        has_cache = past_key_values is not None
+        has_static_cache = isinstance(past_key_values, (StaticCache, HybridCache))
 
         for idx, iter_attentions in enumerate(attentions):
-            tgt_len = min_length + idx if not use_cache else 1
-            src_len = min_length + idx
+            tgt_len = min_length + idx if not has_cache else 1
+            src_len = min_length + idx if not has_static_cache else past_key_values.get_max_cache_shape()
```

Comment:
I had trouble to understand this part (but it's not the new change, I don't understand well what happens here even in the current `main`)

Do you think a bit of comment in the code will help future readers?

Reply:
These are indeed poor old variable names!

Made the following name updates:
`src_len` -> `query_length`
`tgt_len` -> `model_input_length`
`min_length` -> `prompt_length`
`max_length` -> `output_length`
`idx` -> `generated_length`

I've also resolved other ambiguous names (`seq_len` could mean different things in different places) and slightly rewrote other bits to improve clarity.

I'll ping you again when it's ready for a re-review 🤗